tests/test_loop.py

```diff
@@ -43,3 +43,24 @@ def test_default_loop(self) -> None:
 
 def test_default_loop(loop: Any) -> None:
     assert asyncio.get_event_loop() is loop
+
+
+@pytest.mark.xfail(not PY_38, reason="ThreadedChildWatcher is only available in 3.8+")
+def test_setup_loop_non_main_thread() -> None:
+    child_exc = None
+
+    def target() -> None:
+        try:
+            with loop_context() as loop:
+                assert asyncio.get_event_loop() is loop
+                loop.run_until_complete(test_subprocess_co(loop))
```

Comment:
I see this calls subprocesses but what about registering signal handlers? This is what actually fails on old watchers.

Reply:
Registering signal handlers will always fail in non-main thread, regardless of whether we are using old or new watcher. The new watcher doesn't register signal handlers to avoid this issue.

What we want to ensure in this test is that child processes can be watched (using waitpid as alternative), not whether signals can be successfully registered.

Am I misunderstanding something?

Reply:
Yes. The original problem with xdist was exactly the fact that there's a problem with the signal handlers that happens in an inconsistent manner. This is what blocks https://github.com/aio-libs/aiohttp/pull/5431 and all the previous attempts. I was sure that the new watchers were supposed to fix this based on what @asvetlov mentioned to me privately. Are you sure it's still problematic?

Reply:
What I'm saying is signal handling is an independent problem from watching child process. The old watcher uses signal handling and will crash when mixed with xdist. The new watcher doesn't use signal handling and therefore is not problematic.

For reference, here's cpython's implementation for `ThreadedChildWatcher` https://github.com/python/cpython/blob/3.8/Lib/asyncio/unix_events.py#L1296. It starts a new thread and calls `waitpid`.

Compare that with `SafeChildWatcher` which inherits from `BaseChildWatcher` https://github.com/python/cpython/blob/3.8/Lib/asyncio/unix_events.py#L927. It tries to register signal handler and hence will raise an exception in non-main thread.